### PR TITLE
[onert] Fix btopolSortOperations

### PR DESCRIPTION
**Pull request by Aeren1564**
This commit fixes TrainableGraph btopolSortOperations.

ONE-DCO-1.0-Signed-off-by: YongHyun An <yonghyunz.an@samsung.com>

### CODE REVIEW

**Comment by Aeren1564** [comment]
The model on https://github.com/Samsung/ONE/issues/12325#issuecomment-1916150595 no more produces an invalid topological order.

File: runtime/onert/core/src/ir/train/TrainableGraph.test.cc [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  tgraph.addOutput({output});
+
+  addElementwiseActivationOperation(tgraph, {input}, {y_pred});
+  addLossOperation(tgraph, {y_pred, y_true}, {output});
+
+  EXPECT_NO_THROW(tgraph.topolSortOperations());
+  EXPECT_NO_THROW(tgraph.btopolSortOperations());
+}
+
+TEST(TrainableGraph, topological_sort_nonlinear)
```
Comment (Aeren1564) [inline runtime/onert/core/src/ir/train/TrainableGraph.test.cc:87]: This is a regression test for nonlinear graph

File: runtime/onert/core/src/ir/train/TrainableGraph.test.cc [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  addElementwiseActivationOperation(tgraph, {split}, {lhs});
+  addElementwiseActivationOperation(tgraph, {split}, {rhs});
+  addAddOperation(tgraph, {lhs, rhs}, {y_pred});
+  addLossOperation(tgraph, {y_pred, y_true}, {output});
+
+  EXPECT_NO_THROW(tgraph.topolSortOperations());
+  EXPECT_NO_THROW(tgraph.btopolSortOperations());
+}
+
+TEST(TrainableGraph, neg_topological_sort_cycle)
```
Comment (Aeren1564) [inline runtime/onert/core/src/ir/train/TrainableGraph.test.cc:126]: This test checks if a graph with cycle is correctly rejected.

**Review by jyoungyun — APPROVED** [review]
LGTM Thank you!

**Review by hseok-oh — APPROVED** [review]
LGTM

**Pull request merged**